Add ExternalID support to AWS Auth STS configuration (#26628)
hashicorp/vault: website/content/api-docs/auth/aws.mdx

- `sts_role` `(string: <required>)` - AWS ARN for STS role to be assumed when
  interacting with the account specified. The Vault server must have
  permissions to assume this role.

> - `external_id` `(string: "")` - External ID to be used when assuming
>   the STS role. If set, the STS role must be configured to require this
>   external ID.

Reviewer: ```suggestion
- `external_id` `(string: "")` - The external ID expected by the STS role. The
   associated STS role **must** be configured to require the external ID.
```

Style correction: write in the active voice
